Code/GraphMol/MolInterchange/Writer.cpp

```diff
@@ -230,7 +253,134 @@ void addBond(const Bond &bond, rj::Value &rjBond, rj::Document &doc,
   }
 }
 
-void addConformer(const Conformer &conf, rj::Value &rjConf, rj::Document &doc) {
+template <typename T>
+void addProperties(const T &obj, const std::vector<std::string> &propNames,
+                   rj::Value &properties, rj::Document &doc) {
+  for (const auto &pN : propNames) {
+    rj::Value rjv;
+    try {
+      auto val = obj.template getProp<int>(pN);
+      rjv = val;
+    } catch (const boost::bad_any_cast &) {
+      try {
+        auto val = obj.template getProp<double>(pN);
+        rjv = val;
+      } catch (const boost::bad_any_cast &) {
+        try {
+          auto val = obj.template getProp<std::string>(pN);
+          rjv.SetString(val.c_str(), val.size(), doc.GetAllocator());
+        } catch (const boost::bad_any_cast &) {
+          continue;
```

Comment:
Is it worth throwing a warning message if we cannot cast the property, and end up skipping it?